### ISSUE TITLE
[BUG] DMP is skipping short videos (solved, see opening post)

### ISSUE THREAD

**Issue by Thyrador** [SushiBtw/discord-music-player#231]
**Bug Description**
When trying to play very short videos (<0.40s) DMP seem to skip them, causing them to not play.
If the video is around 41s long, it plays around a second of audio, then skips the video.

**Reproduction Steps**
1. Just applied your code like in the sample
2. Playing the following video (29s) completely skips it: https://www.youtube.com/watch?v=d95PPykB2vE
3. Playing the following video (41s) plays around 1s of audio: https://www.youtube.com/watch?v=BXyl85yH1bo
4. Getting a **The queue has ended.** from _**queueEnd**_ event.

**Expected Result**
All videos being played at full length, even if they are just a few seconds long.

**Versioning:**
- OS: [e.g. Ubuntu 21.04]
- Node Version: v16.11.1
- Module Version: v8.2.0

EDIT:

Solved in https://github.com/SushiBtw/discord-music-player/issues/231#issuecomment-943069710

**Comment by Thyrador** [comment]
> just a quick check what version of DMP are you running. may be related to #219 which was deemed a ytdl issue.

8.2.0 I am running/testing.

And yes, you are right. I have the same issues with both of the videos posted in #219.
Is there any known workaround to this?

**Comment by Night-Hunter-NF** [comment]
@Thyrador  you could try https://github.com/SushiBtw/discord-music-player/issues/219#issuecomment-939615193. it made it sort of work stops it almost always failing to just sometimes failing

**Comment by Thyrador** [comment]
> @Thyrador you could try [#219 (comment)](https://github.com/SushiBtw/discord-music-player/issues/219#issuecomment-939615193). it made it sort of works, stops it almost always failing to just sometimes failing

Tried that already (before opening the issue) without success.It still skips the videos I tested (didn't seem to make much of a difference for me).

**Comment by Thyrador** [comment]
@SushiBtw 
Ha, I fixed it. It was close, but not the right thing to do.

Just do the following:

```js
let stream = discord_ytdl_core_1.default(song.url, {
    requestOptions: (_b = this.player.options.ytdlRequestOptions) !== null && _b !== void 0 ? _b : {},
    opusEncoded: false,
    seek: (options === null || options === void 0 ? void 0 : options.seek) ? options.seek / 1000 : 0,
    fmt: 's16le',
    encoderArgs: [],
    quality: quality.toLowerCase() === 'low' ? 'lowestaudio' : 'highestaudio',
    highWaterMark: 1 << 25,
    filter: 'audioonly' // <== just add this, and it seem to work with all the videos I tested
})
```

**Comment by SushiBtw** [comment]
Let me look into it, also for the first issue - don't pass the quality in data, pass it as the options.

**Comment by Thyrador** [comment]
> Let me look into it, also for the first issue - don't pass the quality in data, pass it as the options.

Thanks for the tip. It does seem to work only if I pass it to the queue, not the song, right?

**Comment by LyrasaurusRose** [comment]
Can confirm the above fix works for me too

**Comment by SushiBtw** [comment]
Fixed by #232, npm push today.

**Comment by SushiBtw** [comment]
Fixed in v8.3.0, pushed per NPM - https://www.npmjs.com/package/discord-music-player/v/8.3.0.
